conda/cli/activate.py

```diff
@@ -24,95 +52,114 @@ def help():
 
 def prefix_from_arg(arg):
     if os.sep in arg:
-        return abspath(arg)
-    prefix = find_prefix_name(arg)
-    if prefix is None:
-        sys.exit('Error: could not find environment: %s' % arg)
+        if isdir(abspath(arg.strip("\""))):
+            prefix = abspath(arg.strip("\""))
+        else:
+            sys.exit('Error: could not find environment: %s' % arg)
+    else:
+        prefix = find_prefix_name(arg)
+        if prefix is None:
+            sys.exit('Error: could not find environment: %s' % arg)
     return prefix
 
 
 def binpath_from_arg(arg):
-    path = join(prefix_from_arg(arg), 'bin')
-    if not isdir(path):
-        sys.exit("Error: no such directory: %s" % path)
+    prefix = prefix_from_arg(arg)
+    if on_win:
+        path = [prefix.rstrip("\\"),
+                join(prefix, 'cmd'),
+                join(prefix, 'Scripts'),
+                join(prefix, 'Library', 'bin'),
+               ]
+    else:
+        path = [prefix.rstrip("/"),
+                join(prefix, 'cmd'),
+                join(prefix, 'bin'),
+                ]
     return path
 
+
+def pathlist_to_str(paths):
+    """
+    Format a path list, e.g., of bin paths to be added or removed,
+    for user-friendly output.
+    """
+    return ' and '.join([path.replace("\\\\", "\\") for path in paths])
```

Comment:
Assuming that your path list contains the normalized paths, this `replace` is not necessary. And it might accidentally hide some internal string inconsistencies.